aptos-move/block-executor/src/executor.rs

```diff
@@ -259,25 +259,26 @@ where
             ));
```

Comment:
Just curious. Why do we need to set `specultive_inconsistent` flag and call the `capture_delayed_field_read_error` function here? We could have directly called this function inside the above `match execute_result` block right?

Reply:
I am doing that in multiple places.

but most relevantly, I was changing that before I had access to read_set. but seems like I've pushed taking read_set to a much earlier space, so I can get rid of it